lsinitrd.sh

```diff
@@ -222,17 +222,30 @@
 }
 
 list_squash_content() {
-    SQUASH_IMG="squash-root.img"
     SQUASH_TMPFILE="$TMPDIR/initrd.root.sqsh"
+    EROFS_TMPMNT="$TMPDIR/erofs.mnt"
 
-    $CAT "$image" 2> /dev/null | cpio --extract --verbose --quiet --to-stdout -- \
-        $SQUASH_IMG > "$SQUASH_TMPFILE" 2> /dev/null
-    if [[ -s $SQUASH_TMPFILE ]]; then
-        echo "Squashed content ($SQUASH_IMG):"
-        echo "========================================================================"
-        unsquashfs -d "squashfs-root" -ll "$SQUASH_TMPFILE" | tail -n +4
-        echo "========================================================================"
-    fi
+    for img in squash-root.img squashfs-root.img erofs-root.img; do
+	$CAT "$image" 2> /dev/null | cpio --extract --verbose --quiet --to-stdout -- \
+	    $img > "$SQUASH_TMPFILE" 2> /dev/null
+	[[ -s $SQUASH_TMPFILE ]] && break
+    done
+    [[ -s $SQUASH_TMPFILE ]] || return 0
+
+    echo "Squashed content ($img):"
+    echo "========================================================================"
+    case $img in
+	squash-root.img|squashfs-root.img)
+		unsquashfs -ll "$SQUASH_TMPFILE" | tail -n +4
+		;;
+	erofs-root.img)
+		mkdir -p "$EROFS_TMPMNT"
+		mount -t erofs "$SQUASH_TMPFILE" "$EROFS_TMPMNT"
+		(cd $EROFS_TMPMNT; find "$EROFS_TMPMNT" -ls)
```

Comment:
https://github.com/redhat-plumbers/dracut-rhel10/pull/7/commits/6bcfe67d6b19d303cb1c4871c04ef18c74f7550f

Also, I've just realized you are be using tabs instead of spaces :) please use spaces in dracut code.

Reply:
Yeah, I saw the shfmt errors as well. But then it was after 6pm on a Friday and I decided that it's something that can wait till Monday ;-)